cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -459,164 +423,15 @@ def try_load_profiles(file_path: str, loader: Callable) -> List[AzureProfile]:
     return loader(file_path)
 
 
-def azure_login_interactively() -> Optional[AzureAccountLoginInfo]:
-    """
-    Return login information on login success
-    Return None on login error
-    """
-
-    COMMAND = "login --query '[0]'"  # returns a dict
-    output_dict = az_cli(COMMAND)
-    if not isinstance(output_dict, dict):
-        return None
-
-    try:
-        subscription_id, tenant_id = output_dict["id"], output_dict["tenantId"]
-        log.info("Logged into Azure account with subscription_id '%s', tenant_id '%s'", subscription_id, tenant_id)
-        return AzureAccountLoginInfo(subscription_id=subscription_id, tenant_id=tenant_id)
-    except KeyError:
-        log.exception("Failed to retrieve Azure account subscription_id and tenant_id")
-        return None
-
-
-def azure_logout() -> None:
-    az_cli("logout")
-
-
-def setup_service_principal(subscription_id: str, name: str) -> Optional[ServicePrincipal]:
-    """
-    Return ServicePrincipal on success
-    Return None on error
-    """
-
-    principal = new_service_principal(subscription_id, name)
-    if principal is None:
-        return None
-
-    if not add_read_write_all_permissions(principal.id):
-        return None
-
-    if not grant_permissions(principal.id):
-        return None
-
-    if not consent_permissions(principal.id):
-        return None
-
-    return principal
-
-
-def list_locations() -> List[str]:
-    """
-    Return currently available Azure locations on success
-    Return empty list on error
-    """
-
-    COMMAND = "account list-locations --query '[].name'"  # returns a list
-    output_list = az_cli(COMMAND)
-    if not isinstance(output_list, list):
-        log.error("Failed to list Azure Locations")
-        return []
-
-    log.info("Retrieved %d Azure Locations", len(output_list))
-    return sorted(output_list)
-
-
-def list_resource_groups(location: str) -> List[str]:
+def list_resource_groups(client: AzureClient, location: str) -> List[str]:
```

Comment:
I think this function is not necessary. The log message is IMHO not worth it.

Reply:
removed